File: 94. Binary Tree Inorder Traversal/solution_1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution 
{
    public List<Integer> values = new ArrayList<>();
    
    public List<Integer> inorderTraversal(TreeNode root) 
    {
        if (root != null)
        {
            inorderTraversal(root.left);
            values.add(root.val);
            inorderTraversal(root.right);
        }
        
        return values;
    }
}

File: 10-99/94. Binary Tree Inorder Traversal/solution_1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution
{
    public IList<int> values = new List<int>();

    public IList<int> InorderTraversal(TreeNode root)
    {
        if (root != null)
        {
            InorderTraversal(root.left);
            values.Add(root.val);
            InorderTraversal(root.right);
        }

        return values;
    }
}